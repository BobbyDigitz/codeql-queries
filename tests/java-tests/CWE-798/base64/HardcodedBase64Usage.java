import java.util.Base64;

class MyApp {
    public static String KEY = "VGVzdFBhc3N3b3Jk";
    public static byte[] KEY2 = new byte[] { 'V', 'G', 'V', 'z', 'd', 'F', 'B', 'h', 'c', '3', 'N', '3', 'b', '3', 'J',
            'k' };

    public String getDecoderString() {
        byte[] decodedBytes = Base64.getDecoder().decode(MyApp.KEY);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderBytes() {
        byte[] decodedBytes = Base64.getDecoder().decode(MyApp.KEY2);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderConvertString() {
        String key = MyApp.KEY2.toString();
        byte[] decodedBytes = Base64.getDecoder().decode(key);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

    public String getDecoderConvertBytes() {
        byte[] key = MyApp.KEY.getBytes();
        byte[] decodedBytes = Base64.getDecoder().decode(key);

        String decodedString = new String(decodedBytes);
        return decodedString;
    }

}