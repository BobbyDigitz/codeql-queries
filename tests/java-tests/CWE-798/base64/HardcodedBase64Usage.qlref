CWE-798/HardcodedBase64Usage.ql