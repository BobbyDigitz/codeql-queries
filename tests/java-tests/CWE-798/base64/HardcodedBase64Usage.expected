edges
| HardcodedBase64Usage.java:4:26:4:28 | KEY : String | HardcodedBase64Usage.java:9:58:9:66 | MyApp.KEY |
| HardcodedBase64Usage.java:4:26:4:28 | KEY : String | HardcodedBase64Usage.java:31:22:31:30 | MyApp.KEY : String |
| HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" : String | HardcodedBase64Usage.java:4:26:4:28 | KEY : String |
| HardcodedBase64Usage.java:5:26:5:29 | KEY2 : byte[] | HardcodedBase64Usage.java:16:58:16:67 | MyApp.KEY2 |
| HardcodedBase64Usage.java:5:26:5:29 | KEY2 : byte[] | HardcodedBase64Usage.java:23:22:23:31 | MyApp.KEY2 : byte[] |
| HardcodedBase64Usage.java:5:33:6:17 | new byte[] : byte[] | HardcodedBase64Usage.java:5:26:5:29 | KEY2 : byte[] |
| HardcodedBase64Usage.java:23:22:23:31 | MyApp.KEY2 : byte[] | HardcodedBase64Usage.java:24:58:24:60 | key |
| HardcodedBase64Usage.java:31:22:31:30 | MyApp.KEY : String | HardcodedBase64Usage.java:31:22:31:41 | getBytes(...) : byte[] |
| HardcodedBase64Usage.java:31:22:31:30 | MyApp.KEY : String | HardcodedBase64Usage.java:32:58:32:60 | key |
| HardcodedBase64Usage.java:31:22:31:41 | getBytes(...) : byte[] | HardcodedBase64Usage.java:32:58:32:60 | key |
nodes
| HardcodedBase64Usage.java:4:26:4:28 | KEY : String | semmle.label | KEY : String |
| HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" : String | semmle.label | "VGVzdFBhc3N3b3Jk" : String |
| HardcodedBase64Usage.java:5:26:5:29 | KEY2 : byte[] | semmle.label | KEY2 : byte[] |
| HardcodedBase64Usage.java:5:33:6:17 | new byte[] : byte[] | semmle.label | new byte[] : byte[] |
| HardcodedBase64Usage.java:9:58:9:66 | MyApp.KEY | semmle.label | MyApp.KEY |
| HardcodedBase64Usage.java:16:58:16:67 | MyApp.KEY2 | semmle.label | MyApp.KEY2 |
| HardcodedBase64Usage.java:23:22:23:31 | MyApp.KEY2 : byte[] | semmle.label | MyApp.KEY2 : byte[] |
| HardcodedBase64Usage.java:24:58:24:60 | key | semmle.label | key |
| HardcodedBase64Usage.java:31:22:31:30 | MyApp.KEY : String | semmle.label | MyApp.KEY : String |
| HardcodedBase64Usage.java:31:22:31:41 | getBytes(...) : byte[] | semmle.label | getBytes(...) : byte[] |
| HardcodedBase64Usage.java:32:58:32:60 | key | semmle.label | key |
subpaths
#select
| HardcodedBase64Usage.java:9:58:9:66 | MyApp.KEY | HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" : String | HardcodedBase64Usage.java:9:58:9:66 | MyApp.KEY | Sensative data is being logged $@. | HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" | user-provided value |
| HardcodedBase64Usage.java:16:58:16:67 | MyApp.KEY2 | HardcodedBase64Usage.java:5:33:6:17 | new byte[] : byte[] | HardcodedBase64Usage.java:16:58:16:67 | MyApp.KEY2 | Sensative data is being logged $@. | HardcodedBase64Usage.java:5:33:6:17 | new byte[] | user-provided value |
| HardcodedBase64Usage.java:24:58:24:60 | key | HardcodedBase64Usage.java:5:33:6:17 | new byte[] : byte[] | HardcodedBase64Usage.java:24:58:24:60 | key | Sensative data is being logged $@. | HardcodedBase64Usage.java:5:33:6:17 | new byte[] | user-provided value |
| HardcodedBase64Usage.java:32:58:32:60 | key | HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" : String | HardcodedBase64Usage.java:32:58:32:60 | key | Sensative data is being logged $@. | HardcodedBase64Usage.java:4:32:4:49 | "VGVzdFBhc3N3b3Jk" | user-provided value |
